src/com/zone5cloud/retrofit/core/OkHttpClientInterceptor_Authorization.java

```diff
@@ -102,13 +100,18 @@ public void run() {
 						}
 					}
 				});
+				clientConfig.setToken(token);
 			}
```

Comment:
We want the token to be set before calling the delegates. This is what `AuthToken previousValue = this.token.getAndSet(token);` does at the top of this method. It sets the token to the new value, uses the old value for comparison and calls the callbacks if it has changed.

Reply:
I had trouble with this one , there was no relation between authToken and clientConfig authToken is AtomicReference so it is getAndSet atomically . I had to set it ClientConfig in this way since it is an AtomicReference .I am setting the previousValue to ClientConfig on line 90 clientConfig.setToken(previousValue); and then in the end on line 103 (the one above)